mission_control/test/test_jaus_ros_bridge_interface_mission_control.py

```diff
@@ -0,0 +1,136 @@
+ *
+ *  Copyright (c) 2020, QinetiQ, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of QinetiQ nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+"""
+import sys
+import os
+import unittest
+import math
+import rosnode
+import rospy
+import rostest
+from mission_control.msg import LoadMission
+from mission_control.msg import ExecuteMission
+from mission_control.msg import ReportExecuteMissionState
+from mission_control.msg import ReportLoadMissionState
+from mission_control.msg import AbortMission
+from mission_control.msg import QueryMissions
+from mission_control.msg import RemoveMissions
+from mission_control.msg import ReportMissions
+
+
+class TestJausRosBridgeInterface(unittest.TestCase):
+    """ 
+        Simulate messages sent by the jaus ros bridge
+    """
+
+    rospy.init_node('test_mission_control_interface_jaus_ros_bridge')
+    mission_execute_state = ReportExecuteMissionState.PAUSED
+    mission_load_state = False
+    dir_path = os.path.dirname(os.path.abspath(__file__)) + '/test_files/'
+
+    mission_load_state_flag = False
+    report_mission_flag = False
+    report_mission = ReportMissions()
+
+    def callback_mission_execute_state(self, msg):
+        self.report_execute_mission = msg
+
+    def callback_report_mission(self, msg):
+        self.report_mission = msg
+        self.report_mission_flag = True
+
+    def callback_mission_load_state(self, msg):
+        self.mission_load_state = msg.load_state
+        self.mission_load_state_flag = True
```

Comment:
@patapufate nit: why not making using `None` to mark that a given attribute has not been initialized yet?

Reply:
Changed